### PR TITLE
Compuccino/fix live gap handling

### PR DESCRIPTION
**Pull request by tchakabam**
## Description

-> See commit messages.

## Specific Changes proposed
Please list the specific changes involved in this pull request.

## Requirements Checklist
- [ ] Feature implemented / Bug fixed
- [ ] If necessary, more likely in a feature request than a bug fix
  - [ ] Unit Tests updated or fixed
  - [ ] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](http://jsbin.com/liwecukasi/edit?html,output))
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/segment-loader.js [hunk 1]
```diff
@@ -1109,9 +1125,12 @@ export default class SegmentLoader extends videojs.EventTarget {
       return;
     }
 
-    if (segmentInfo.timestampOffset !== null &&
-        segmentInfo.timestampOffset !== this.sourceUpdater_.timestampOffset()) {
-      this.sourceUpdater_.timestampOffset(segmentInfo.timestampOffset);
+    const timelineMapping = this.syncController_.mappingForTimeline(segmentInfo.timeline);
```
Comment (ErfanNaz) [inline src/segment-loader.js:1129]: der syncController hat keine Funktion mappingForTimeline, vielleicht hast du vergessen diese Funktion mit zu commiten ?

**Review by ErfanNaz — COMMENTED** [review]
mappingForTimeline wurde mit commited, daher wird immer ein Fehler geworfen.

File: src/segment-loader.js (hunk 1 above)
Comment (tchakabam) [reply src/segment-loader.js:1129]: you are right :+1:

**Comment by tchakabam** [comment]
now its there

**Comment by ErfanNaz** [comment]
I have test it with the prepared Playlist from Marcus, and it still not working. Which playlist do you use for testing?

**Comment by tchakabam** [comment]
Ok, will check.

I tested with Marcus stream, but with the fix on master, not this one actually. Here i just patched the code.

Let me try this now.

**Comment by tchakabam** [comment]
Hey @illiano, I have now managed to also make the fix work with the older version here. It's just slightly different and actually makes more sense! I tested it with the stream Marcus had sent, and it works 5/5 times :) and also normal test VOD stream works well still.

**Comment by tchakabam** [comment]
:+1:

**Pull request merged**